Extension/src/Debugger/configurationProvider.ts

```diff
@@ -1128,122 +1197,95 @@ export class ConfigurationAssetProviderFactory {
+            new LldbDapConfigurations(this.executable)
         ];
     }
 }
 
-class OSXConfigurationProvider extends DefaultConfigurationProvider {
-    private MIMode: string = 'lldb';
+class OSXConfigurationProvider extends ConfigurationAssetProvider {
+
     private executable: string = "a.out";
-    private pipeProgram: string = "/usr/bin/ssh";
 
     constructor() {
         super();
         this.configurations = [
-            new MIConfigurations(this.MIMode, this.executable, this.pipeProgram)
+            new LldbDapConfigurations(this.executable),
+            new MIConfigurations('lldb', this.executable)
         ];
     }
 }
 
-class LinuxConfigurationProvider extends DefaultConfigurationProvider {
-    private MIMode: string = 'gdb';
-    private setupCommandsBlock: string = `"setupCommands": [
-    {
-        "description": "${localize("enable.pretty.printing", "Enable pretty-printing for {0}", "gdb").replace(/"/g, '')}",
-        "text": "-enable-pretty-printing",
-        "ignoreFailures": true
-    },
-    {
-        "description": "${localize("enable.intel.disassembly.flavor", "Set Disassembly Flavor to {0}", "Intel").replace(/"/g, '')}",
-        "text": "-gdb-set disassembly-flavor intel",
-        "ignoreFailures": true
-    }
-]`;
+class LinuxConfigurationProvider extends ConfigurationAssetProvider {
+    private additionalProperties = {
+        setupCommands: [
+            {
+                description: localize("enable.pretty.printing", "Enable pretty-printing for {0}", "gdb").replace(/"/g, ''),
+                text: "-enable-pretty-printing",
+                ignoreFailures: true
+            },
+            {
+                description: localize("enable.intel.disassembly.flavor", "Set Disassembly Flavor to {0}", "Intel").replace(/"/g, ''),
+                text: "-gdb-set disassembly-flavor intel",
+                ignoreFailures: true
+            }
+        ]
+    };
     private executable: string = "a.out";
     private pipeProgram: string = "/usr/bin/ssh";
 
     constructor() {
         super();
         this.configurations = [
-            new MIConfigurations(this.MIMode, this.executable, this.pipeProgram, this.setupCommandsBlock),
-            new PipeTransportConfigurations(this.MIMode, this.executable, this.pipeProgram, this.setupCommandsBlock)
+            new MIConfigurations('gdb', this.executable, this.additionalProperties),
+            new PipeTransportConfigurations('gdb', this.executable, this.pipeProgram, this.additionalProperties),
+            new LldbDapConfigurations(this.executable)
         ];
     }
 }
 
 function convertConfigurationSnippetToCompletionItem(snippet: IConfigurationSnippet): vscode.CompletionItem {
     const item: vscode.CompletionItem = new vscode.CompletionItem(snippet.label, vscode.CompletionItemKind.Module);
 
-    item.insertText = snippet.bodyText;
+    item.insertText = JSON.stringify(snippet.body, null, '\t');
```

Comment:
Does VS Code not already format the JSON for us? I would hope we could avoid this if possible. Otherwise, you might need to lookup the indentation preferences for JSON files from the user.